pkg/scanner/scanner.go

```diff
@@ -0,0 +1,171 @@
+package scanner
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/fluxcd/pkg/version"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	"github.com/go-logr/logr"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	"github.com/weaveworks/profiles/pkg/gitrepository"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+//go:generate counterfeiter -o fakes/fake_git_client.go . GitClient
```

Comment:
do you get warnings from counterfeiter when you generate with this?

they recommend now to have this line once per package:
```go
//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
```

then lines like this per interface:
```go
//counterfeiter:generate -o fakes/fake_thing.go . FooBar
```

Reply:
I didn't notice any warnings :shrug:

Reply:
huh 🤷‍♀️  maybe your counterfeiter is old?

Reply:
fixed! Got the warnings when I updated, good spot!